### DISCUSSION TITLE
question about renaming sensors

### DISCUSSION THREAD

**Discussion by jmporter34** [dgreif/ring#881] [Q&A] [answered]
First, thank you @dgreif for your work on this.  I can now see my Ring Alarm and all its associated sensors in both Homebridge and Home Assistant for the first time!

My question is this.  Now that I can see all the sensor names on one screen, I noticed that I've moved some of the sensors from their original locations in my home to new locations, but their names in the Ring app still reflected their original locations.  For a test, I renamed one in the Ring app to reflect its new location.  I was hoping that Homebridge might pick up the new sensor name automatically, but that didn't happen.  I assume that if I were to uninstall and then reinstall your plugin, it would get the latest sensor names.  I'm a little leery of doing that, and was just wondering if there's any other way to force a refresh of the sensor names.

Thanks again!

**Comment by dgreif** [comment, marked as answer]
Re-installing _would_ pick up those changes.  I don't think there is a way for the plugin to reflect those changes, since they can be renamed in HomeKit and it keeps track of those renaming events on it's own.

**Reply by jmporter34** [reply]
I tried uninstalling and re-installing the Ring plugin but it still shows the old sensor names.  Maybe I have to remove each sensor from the Ring mobile app (to force removal from the Ring cloud database), and then add each back in the Ring app with the new names.  And then uninstall and re-install the Ring plugin.  Ughhh...
Anyway, thanks for getting back to me.

**Reply by dgreif** [reply]
Hmm it's possible the names were cached in HomeKit after removing the plugin. Sorry it's such a heads ache!

**Reply by mparsh70** [reply]
I had same issue but “was” able to rename by going into “bridge” on Security System on HomeKit. However figured out name is constrained to 15 chars so for me “Garag Entry door” had to omit “e” otherwise renamed to just “Entry door”.   But was able to rename.   Is there a way to see the accessories in HomeKit without having to dig into the Security System?  Would like to see right away status of all my doors”contact sensors” at night. Thx love the plug-in!

**Reply by dgreif** [reply]
@mparsh70 each accessory (contact sensor, motion sensor, etc) should be exposed as a top-level accessory in HomeKit.  If you are seeing something different, maybe you could post a screenshot?

**Reply by mparsh70** [reply]
I’ll see if I can attach my note here. I can see the sensors I just seem to have to navigate 4 levels deep on the Alarm to see them vs right on the main page. 
![AF6FCBB2-B798-4E95-8191-14DD5AF14191](https://user-images.githubusercontent.com/34952589/155537501-d90561b7-642c-480f-adab-936ba0f59bbe.jpeg)
![E77ABE57-A466-4EBC-8746-E4B38644BE36](https://user-images.githubusercontent.com/34952589/155537552-02eed70a-ce99-40a8-a765-27e13cca7d33.jpeg)

**Reply by mparsh70** [reply]
![399B4C67-8F5D-431A-B1C9-4998F12B76AD](https://user-images.githubusercontent.com/34952589/155537667-bb85d435-32e7-4410-b90a-e60398cdf2cc.jpeg)

**Reply by dgreif** [reply]
@mparsh70 that's just the way the Home app groups the sensors.  I think the idea is that you generally don't need to see individual sensors at a high level. Not something I can change from the plugin tho, sorry!

**Comment by jmporter34** [comment]
@mparsh70 So you're using the Apple Home app, and changed the sensor names there, right ?  Did the new names appear in Homebridge as well, or still the old names there ?  
@dgreif 
I've ditched Apple Home and am using Home Assistant as an alternative.  But in either case, I was under the impression that the Homebridge Ring plugin gets the sensor names from Ring, and the plugin then tells either Apple Home or Home Assistant what the sensor names are - not that Apple Home or Home Assistant can tell the plugin what the sensor names are.
When I changed my sensor names in the Ring mobile app (and verified that Ring knows about the new names by logging into ring.com on my laptop), the old names still appear in both Homebridge and Home Assistant, even after uninstalling and re-installing the Homebridge Ring plugin.  My next step is a complete reinstall of Homebridge on my Raspberry Pi.  If I still see the old sensor names after doing that, I may tear out what's left of my hair!

**Reply by dgreif** [reply]
I think the names probably stick around in the `persist` folder for homebridge.  That's all beyond the plugin, in homebridge/HomeKit land

**Reply by jmporter34** [reply]
Oh, I'm new to Homebridge and hadn't heard about the "persist" folder.  Is it safe to delete that ?

**Reply by dgreif** [reply]
It will cause all of your devices to be fully reset, including any automations or room assignments you have in HomeKit.  Not safe to delete, but is something you can delete if you really want to reset everything

**Reply by jmporter34** [reply]
I hadn't yet set up any automations or room assignments in Home Assistant for the Ring devices, so nothing to lose.  In Homebridge, I ran a terminal and deleted the persist folder, rebooted, and still saw the old sensor names.  Then I uninstalled the Ring plugin, ran a terminal and deleted the persist folder again, rebooted again, re-installed the Ring plugin, and STILL saw the old sensor names.  Jeez!
Finally I re-imaged my Homebridge SD card and started from scratch installing Homebridge and then the Ring plugin.  Finally, the new sensor names appear in Homebridge!  Then I removed the old Homebridge Homekit Controller from Home Assistant, and added the new one.  And I can see the new sensor names there as well!  Huzzah!
One odd thing, when I configure the Homebridge Homekit Controller in Home Assistant, I get this weird "invalid flow" message popping up.  It happened the first time I did this a couple of weeks ago too.  But it apparently can be ignored, because the Homebridge/Ring devices are all visible in Home Assistant.
Thanks for your suggestions, and like I said up front, for your work that makes it possible for me to see my Ring devices in Home Assistant.  Do you have one of those "buy me a beer/coffee" things set up ?

**Reply by dgreif** [reply]
There is a "sponsor" button on the repo if you are feeling generous. Glad you got it working!
